### PR TITLE
[SPARKTA-361] Sparta with SSL

### PR DESCRIPTION
**Pull request by anistal**
#### Description
As an user I want to configure a certificate in my configuration as follows:

```
spray.can {
    server {
      server-header = "sparta api"
      ssl-encryption = "on"
      ssl-tracing = "on"
      verbose-error-messages = "on"
    }
}

sparta {
  api {
    host = localhost
    port = 9090
    certificate-file = "/home/user/certificates/stratio.jks"
    certificate-password = "certificate password"
  }

  ......
```
Then sparta has ssl support: https://localhost:9090 and all requests are securized.

To generate a valid certification:
```
keytool -genkey -keyalg RSA -alias stratio -keystore stratio.jks -storepass stratio -validity 360 -keysize 2048
```

### CODE REVIEW

File: serving-api/src/main/resources/reference.conf [hunk 1]
```diff
@@ -68,8 +70,8 @@ sparta {
     numExecutors = 2
     executorMemory = 2G
     executorCores = 2
-    spark.app.name = SPARTA
-    #spark.metrics.conf = /opt/sds/sparta/benchmark/src/main/resources/metrics.properties
+    spark.app.name = SPARKTA
+    #spark.metrics.conf = /opt/sds/sparkta/benchmark/src/main/resources/metrics.properties
   }
 
   standalone {
```
Comment (danielcsant) [inline serving-api/src/main/resources/reference.conf]: Revert the rename changes

**Comment by danielcsant** [comment]
Please add the configuration also at the end of the configuration file `application.conf`

**Comment by danielcsant** [comment]
Add documentation of this feature

File: serving-api/src/main/resources/reference.conf [hunk 2]
```diff
@@ -30,7 +23,7 @@ sparta {
     spark.master = "local[*]"
     spark.cores.max = 4
     spark.executor.memory = 1024m
-    spark.app.name = SPARTA
+    spark.app.name = SPARKTA
```
Comment (danielcsant) [inline serving-api/src/main/resources/reference.conf]: Please revert this

**Comment by anistal** [comment]
Done documentation.

**Comment by danielcsant** [comment]
+1!

**Comment by acemilio** [comment]
Above, within your description, you talk about several properties:

spray.can {
    server {
      server-header = "sparta api"
      ssl-encryption = "on"
      ssl-tracing = "on"
      verbose-error-messages = "on"
    }
}

but in reference.conf and application.conf appears  **ssl-encryption = "on"**  property only. Is that correct?? Are all mandatory??

Anyway... +1

**Comment by anistal** [comment]
Only it is mandatory ssl-encryption = "on"

**Comment by antonioalf** [comment]
:shipit:

**Comment by stratiocommit** [comment]
Comment accepted. Will treat it ASAP

**Comment by stratiocommit** [comment]
Squashing and checks in progress.

**Pull request merged**